kiota_http/middleware/redirect_handler.py

```diff
@@ -168,6 +174,13 @@ def _redirect_url(self, request: httpx.Request, response: httpx.Response) -> htt
         except Exception as exc:
             raise Exception(f"Invalid URL in location header: {exc}.")
 
+        if url.scheme != request.url.scheme and not options.allow_redirect_on_scheme_change:
```

Comment:
Maybe consider allowing an upgrade from http to https? It sometimes happens with CDNs, and some people publish static generated APIs

Reply:
Makes sense. Should we make this the default behavior across the http packages? For instance dotnet doesn't support that: https://github.com/microsoft/kiota-http-dotnet/blob/119e79cf50dcb94770ba3942ff13e1e9ab57bf8e/src/Middleware/RedirectHandler.cs#L130C1-L131C1

Reply:
I don't have all the context for that original implantation but maybe @andrueastman does ?
Other than that I'd say yes.

Reply:
The origin of this was to initially prevent downgrading to a less secure protocol (e.g https to http) for security reasons as the goal was to be secure by default and let the user opt in to sending/receiving data to/from a potentially unsecure endpoint. https://github.com/microsoft/kiota/pull/589#discussion_r703752506

Given that this behavior is still technically configurable through the `ShouldRedirect` lambda(can technically be used to whitelist schemes) and the `allow_redirect_on_scheme_change`, I don't think we need to change anything here for now and probably handle the discussion separately as changing this now will probably introduce code that checks and compares secure and insecure schemes that are valid for upgrades(unless http to https is the only scenario here?).

Reply:
Thanks for the additional information!
http to https only scenario: yes as we depend on an http client under the hood anyway. Some "http clients" used to be able to do FTP/SFTP or SMB, but I think most of those have disappeared in favor of protocol specific clients over time.

Handling this in a different issue to unblock this PR: sure! @samwelkanda can you create an issue in the sdk design repo to update the spec so it includes that aspect please?